virtcontainers/pkg/oci/utils.go

```diff
@@ -222,7 +222,81 @@ func newLinuxDeviceInfo(d spec.LinuxDevice) (*config.DeviceInfo, error) {
 	return &deviceInfo, nil
 }
 
-func containerDeviceInfos(spec CompatOCISpec) ([]config.DeviceInfo, error) {
+func configContainsPCIColdplugGroup(hConfig *vc.HypervisorConfig, iommuGroup string) bool {
+	for _, g := range hConfig.PCIColdplugGroups {
+		if iommuGroup == strings.TrimSpace(g) {
+			return true
+		}
+	}
+	return false
+}
+
+// getBDF returns the BDF of pci device
+// Expected input strng format is [<domain>]:[<bus>][<slot>].[<func>] eg. 0000:02:10.0
+func getBDF(deviceSysStr string) (string, error) {
+	tokens := strings.Split(deviceSysStr, ":")
+
+	if len(tokens) != 3 {
+		return "", fmt.Errorf("Incorrect number of tokens found while parsing bdf for device : %s", deviceSysStr)
```

Comment:
I'd consider making the `3` a `const` and adding that to the returned error so we log `3` vs. `len(tokens)`.